nf_core/modules/lint/main_nf.py

```diff
@@ -236,25 +236,30 @@ def check_process_section(self, lines, fix_version, progress_bar):
 
     # Check that process labels are correct
     correct_process_labels = ["process_single", "process_low", "process_medium", "process_high", "process_long"]
-    process_label = [l for l in lines if l.lstrip().startswith("label")]
-    if len(process_label) > 0:
-        try:
-            process_label = re.search("process_[A-Za-z]+", process_label[0]).group(0)
-        except AttributeError:
-            process_label = re.search("'([A-Za-z_-]+)'", process_label[0]).group(0)
-        finally:
-            if not process_label in correct_process_labels:
+    all_labels = [l for l in lines if l.lstrip().startswith("label ")]
```

Comment:
It might be a good idea to extract this into a function. Something like `get_process_label(s)`. That would make things more readable and testable. And also have tests.

Reply:
The whole method is now getting quite bloated. If the code here is accepted then I think a next step could be to split out the whole label section rather than just splitting out the list comprehension. Similarly it may be worth considering splitting out the container parsing part.

Reply:
That was then less clear than I hoped. I meant taking out the whole label selection as you propose - not just a list comprehension.

Reply:
And I think it would be ideal to do it as part of this PR.